### PR TITLE
Issue #186 more basic IP tools

### PR DESCRIPTION
**Pull request by shamirShahzad**
Finished adding all the basic IP Tools which include:

is_loopback
is_multicast
is_unspecified
is_global_unicast
is_interface_local_multicast
is_link_local_multicast
is_link_local_unicast

All new .go files can be found in the ipinfo and lib directories with the file name cmd_tool_NAME OF TOOL also the cmd_tool.go file was modified to allow access to the tools via the cli

### CODE REVIEW

File: ipinfo/cmd_tool_is_global_unicast.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+`, progBase)
+}
+
+func cmdToolisGlobalUnicast()(err error){
+	f:=lib.CmdToolIsGlobalUnicastFlags{}
+	f.Init()
+	pflag.Parse()
+
+	return lib.CmdToolIsGlobalUnicast(f,pflag.Args()[2:],printHelpToolIsGlobalUnicast)
+}
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is_global_unicast.go:56]: Put an ending newline in all files

File: ipinfo/cmd_tool.go [hunk 2]
```diff
@@ -11,21 +11,28 @@ import (
...
+		"n2ip":                        completionsToolN2IP,
+		"n2ip6":                       completionsToolN2IP6,
+		"prefix":                      completionsToolPrefix,
+		"is_loopback":                 completionsToolIsLoopBack,
+		"is_multicast":                completionsToolIsMulticast,
+		"is_unspecified":              completionsToolIsUnspecified,
+		"is_global_unicast":           completionsToolIsGlobalUnicast,
+		"is_link_local_unicast":       completionsToolIsLinkLocalUnicast,
+		"is_link_local_multicast":     completionsToolIsLinkLocalMulticast,
+		"is_interface_local_multicast":completionsToolIsInterfaceLocalMulticast,
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:35]: Did you run the formatter?

File: ipinfo/cmd_tool.go [hunk 3]
```diff
@@ -39,21 +46,28 @@ func printHelpTool() {
...
+  n2ip	                         evaluates a mathematical expression and converts it to an IPv4 or IPv6.
+  n2ip6	                         evaluates a mathematical expression and converts it to an IPv6.
+  prefix                         misc. prefix tools related to CIDRs.
+  is_loopback                    reports whether an IP is a valid Loopback address.
+  is_multicast                   reports whether an IP is a valid Multicast address.
+  is_unspecified                 reports whether an IP is an Unspecified address.
+  is_global_unicast              reports whether an IP is a global unicast address.
+  is_link_local_unicast          reports whether IP is a Link Local Unicast
+  is_link_local_multicast        reports whether IP is a Link Local Multicast address
+  is_interface_local_multicast   reports whether an IP is a interface local multicast
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:70]: Can delete 2 spaces here and in all the ones above

File: ipinfo/cmd_tool.go [hunk 4]
```diff
@@ -96,6 +110,20 @@ func cmdTool() error {
...
+	case cmd == "is_unspecified":
+		err = cmdToolIsUnspecified()
+	case cmd == "is_global_unicast":
+		err = cmdToolisGlobalUnicast()
+	case cmd == "is_interface_local_multicast":
+		err = cmdToolIsInterfaceLocalMulticast()
+	case cmd == "is_link_local_multicast":
+		err = cmdToolIsLinkLocalMulticast()
+	case cmd == "is_link_local_unicast":
+		err = cmdToolIsLinkLocalUnicast()
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:126]: Please align this to be equal in order to the command list itself in the `-h`

**Review by UmanShahzad — COMMENTED** [review]
You didnt run the fmt'er yet

**Comment by shamirShahzad** [comment]
@UmanShahzad kindly review

File: ipinfo/cmd_tool.go [hunk 5]
```diff
@@ -39,21 +46,28 @@ func printHelpTool() {
...
+  n2ip	                       evaluates a mathematical expression and converts it to an IPv4 or IPv6.
+  n2ip6	                       evaluates a mathematical expression and converts it to an IPv6.
+  prefix                       misc. prefix tools related to CIDRs.
+  is_loopback                  reports whether an IP is a valid Loopback address.
+  is_multicast                 reports whether an IP is a valid Multicast address.
+  is_unspecified               reports whether an IP is an Unspecified address.
+  is_global_unicast            reports whether an IP is a global unicast address.
+  is_link_local_unicast        reports whether IP is a Link Local Unicast
+  is_link_local_multicast      reports whether IP is a Link Local Multicast address
+  is_interface_local_multicast reports whether an IP is a interface local multicast
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool.go:70]:
```suggestion
  is_loopback                  reports whether an IP is a valid loopback address.
  is_multicast                 reports whether an IP is a valid multicast address.
  is_unspecified               reports whether an IP is an unspecified address.
  is_global_unicast            reports whether an IP is a global unicast address.
  is_link_local_unicast        reports whether IP is a link local unicast address.
  is_link_local_multicast      reports whether IP is a link local multicast address.
  is_interface_local_multicast reports whether an IP is a interface local multicast.
```

File: ipinfo/cmd_tool_is_global_unicast.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		"--quiet": predict.Nothing,
+	},
+}
+
+func printHelpToolIsGlobalUnicast() {
+	fmt.Printf(`
+%s tool is_global_unicast [<opts>] <cidr | ip | ip-range | filepath>
+
+Description: Checks the provided address is a Global Unicast Address
+Inputs can be IPs, IP ranges, CIDRs, or filepath to a file
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is_global_unicast.go:26]: Please get the spacing right

File: ipinfo/cmd_tool_is_global_unicast.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+		"-h":      predict.Nothing,
+		"--help":  predict.Nothing,
+		"-q":      predict.Nothing,
+		"--quiet": predict.Nothing,
+	},
+}
+
+func printHelpToolIsGlobalUnicast() {
+	fmt.Printf(`
+%s tool is_global_unicast [<opts>] <cidr | ip | ip-range | filepath>
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is_global_unicast.go:23]: Why is this on its own newline? Please follow all other existing CLI patterns

File: ipinfo/cmd_tool_is_loopback.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  $ %[1]s tool is_loopback 154.0.0.1/32 | ff02::1/64
+
+  # Check for file.
+  $ %[1]s tool is_loopback /path/to/file.txt 
+
+  # Check entries from stdin.
+  $ cat /path/to/file1.txt | %[1]s tool is_loopback
+
+  Options:
+  --help, -h
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is_loopback.go:48]: All the spacing is messed up

**Review by UmanShahzad — CHANGES_REQUESTED** [review]
Please make 100% sure the help style matches the style of all other existing commands. Too much back-and-forth on this.

File: lib/cmd_tool_is_global_unicast.go [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+	ip := net.ParseIP(input)
+	isGlobalUnicast := ip.IsGlobalUnicast()
+
+	fmt.Printf("%s,%v\n", input, isGlobalUnicast)
+}
+
+func ActionIsGlobalUnicastRange(input string) {
+	ipRange, err := IPRangeStrFromStr(input)
+	if err != nil {
+		return
```
Comment (UmanShahzad) [inline lib/cmd_tool_is_global_unicast.go:68]: In is_v4 / is_v6 I see we do atleast print an error.

**Review by UmanShahzad — COMMENTED** [review]
- Command examples are not real it seems. Please make sure you run each command, including your own examples, and that they work and produce **the expected results**!
- Error handling? I guess we can copy the is_v4/is_v6 style for now where it's printed. See those files.

File: lib/cmd_tool_is_global_unicast.go (hunk 9 above)
Comment (shamirShahzad) [reply lib/cmd_tool_is_global_unicast.go:68]: I did have them before but was instructed to get them removed

File: ipinfo/cmd_tool_is_global_unicast.go [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+	fmt.Printf(
+		`Usage: %s tool is_global_unicast [<opts>] <cidr | ip | ip-range | filepath>
+
+Description:
+  Checks if the input is a global unicast address.
+  Inputs can be IPs, IP ranges, CIDRs, or filepath to a file
+
+Examples:
+  $ %[1]s tool is_global_unicast 10.255.0.0 | 2000::1
+  $ %[1]s tool is_global_unicast 255.255.255.255 | ff00::1
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is_global_unicast.go:31]: Still have false examples

**Review by UmanShahzad — CHANGES_REQUESTED** [review]

File: ipinfo/cmd_tool_is_global_unicast.go (hunk 10 above)
Comment (shamirShahzad) [reply ipinfo/cmd_tool_is_global_unicast.go]: sorry about that just forgot to remove the examples the rest should be okay now please check @UmanShahzad

File: ipinfo/cmd_tool_is_link_local_multicast.go [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+Examples:
+  $ %[1]s tool is_link_local_multicast 224.0.0.0
+  $ %[1]s tool is_link_local_multicast 169.200.0.0
+  $ %[1]s tool is_link_local_multicast ff02::2
+  $ %[1]s tool is_link_local_multicast fe80::
+
+  # Check CIDR.
+  $ %[1]s tool is_link_local_multicast 224.0.0.0/32 | ff02::1/64
+  $ %[1]s tool is_link_local_multicast 169.200.0.0/32 | fe80::1/64
```
Comment (UmanShahzad) [inline ipinfo/cmd_tool_is_link_local_multicast.go:37]: ...

**Review by UmanShahzad — CHANGES_REQUESTED** [review]

**Comment by shamirShahzad** [comment]
@UmanShahzad

**Review by UmanShahzad — APPROVED** [review]

File: ipinfo/cmd_tool_is_global_unicast.go (hunk 10 above)
Comment (Shayxxxxx) [reply ipinfo/cmd_tool_is_global_unicast.go]: Talia

**Pull request merged**